lib/core/src/sdk.rs

```diff
@@ -788,11 +788,54 @@ impl LiquidSdk {
                 };
                 payment_destination = SendDestination::Bolt11 { invoice };
             }
-            _ => {
+            Ok(_) => {
                 return Err(PaymentError::Generic {
                     err: "Destination is not valid".to_string(),
                 });
             }
+            Err(_) => {
+                // TODO Workaround to avoid adding an InputType::Bolt12 variant in sdk-common
+
+                // TODO TBD: Should InputType::Bolt12_Offer be added to sdk-common? Or Bolt12_Invoice?
+                //      If yes, with full parsing and mapping to LNInvoice?
+
+                let amount_sat = req
+                    .amount_sat
+                    .ok_or_else(|| anyhow!("Expected amount when processing BOLT12 offer"))?;
+
+                let offer = &req.destination;
+                info!("Got BOLT12 offer, fetching BOLT12 invoice");
+
+                let invoice_str = self.swapper.get_bolt12_invoice(offer, amount_sat)?;
+                let invoice_parsed = utils::parse_bolt12_invoice(&invoice_str)?;
+
+                // TODO If supporting receive: validate it's not a self-transfer
+                // TODO Validate invoice
+
+                receiver_amount_sat = invoice_parsed.amount_msats() / 1_000;
+
+                if let Some(amount_sat) = req.amount_sat {
+                    ensure_sdk!(
+                        receiver_amount_sat == amount_sat,
+                            PaymentError::Generic { err: "Amount in the payment request is not the same as the one in the invoice".to_string() }
+                        );
+                }
+
+                let lbtc_pair = self.validate_submarine_pairs(receiver_amount_sat)?;
+
+                // TODO Check for MRH for Bolt12
+                // TODO If MRH, fallback to onchain
+
+                let boltz_fees_total = lbtc_pair.fees.total(receiver_amount_sat);
+                let lockup_fees_sat = self
+                    .estimate_lockup_tx_fee_send(receiver_amount_sat + boltz_fees_total)
+                    .await?;
+                fees_sat = boltz_fees_total + lockup_fees_sat;
+
+                payment_destination = SendDestination::Bolt12 {
+                    offer: req.destination.to_string(),
```

Comment:
Perhaps we should populate here the bolt12 here instead of the offer? This way also the send_payment can pay directly the bolt12 without fetching again it from boltz.

Reply:
Actually on second thought why do we need to fetch the bolt12 here in the prepare?

Reply:
Done, removed invoice fetching from prepare.